src/app/newSettings/Sections/Workspace/Logs/hooks/useAccessLogs.ts

```diff
@@ -0,0 +1,76 @@
+import { WorkspaceLogOrderBy, WorkspaceLogResponse, WorkspaceLogType } from '@internxt/sdk/dist/workspaces';
+import errorService from 'app/core/services/error.service';
+import workspacesService from 'app/core/services/workspace.service';
+import { RootState } from 'app/store';
+import { useAppSelector } from 'app/store/hooks';
+import { useState, useEffect } from 'react';
+
+export const ACCESS_LOGS_DEFAULT_LIMIT = 20;
+
+interface UseAccessLogsProps {
+  lastDays?: number;
+  member?: string;
+  activity?: WorkspaceLogType[];
+  orderBy?: string;
+}
+
+export const useAccessLogs = ({ activity, lastDays, member, orderBy }: UseAccessLogsProps) => {
+  const selectedWorkspace = useAppSelector((state: RootState) => state.workspaces.selectedWorkspace);
+  const workspaceId = selectedWorkspace?.workspace?.id;
+
+  const [accessLogs, setAccessLogs] = useState<WorkspaceLogResponse[]>([]);
+  const [hasMoreItems, setHasMoreItems] = useState<boolean>(true);
+  const [offset, setOffset] = useState<number>(0);
+  const [isLoading, setIsLoading] = useState(false);
```

Comment:
I thinkin this case we can put it all in a single state to avoid inconsistencies when updating different states at the same time